src/main.ts

```diff
@@ -230,6 +236,21 @@ export class Purchases {
     validateCurrency(params?.currency);
     const appUserId = this._appUserId;
     const offeringsResponse = await this.backend.getOfferings(appUserId);
+
+    if (params?.offeringIdentifier) {
```

Comment:
I guess you can skip this first level if, given that you do the one at line 246.